File: Linked_List/python/linked_list_delete.py
# Python Implementation
# Linked list Delete


# Create a node
class Node:
    def __init__(self, item):
        self.item = item
        self.next = None


class LinkedList:

    def __init__(self):
        self.head = None

    # Insert
    def insert(self, data):
        new_node = Node(data)
        if self.head is None:
            self.head = new_node
            return
        last = self.head
        while (last.next):
            last = last.next
        last.next = new_node



    # Delete
    def deleteNode(self, position):
        if self.head == None:
            return
        temp_node = self.head
        if position == 0:
            self.head = temp_node.next
            temp_node = None
            return
        for i in range(position - 1):
            temp_node = temp_node.next
            if temp_node is None:
                break
        if temp_node is None:
            return
        if temp_node.next is None:
            return
        next = temp_node.next.next
        temp_node.next = None
        temp_node.next = next

    def printList(self):
        temp_node = self.head
        while (temp_node):
            print(str(temp_node.item) + " ", end="")
            temp_node = temp_node.next


if __name__ == '__main__':
    
    llist = LinkedList()
    
    print('Linked list:')
    llist.printList()
    
    
    llist.insert(5)
    llist.insert(2)
    llist.insert(3)
    llist.insert(6)
    llist.insert(7)
    llist.insert(1)
    llist.insert(9)
    llist.insert(10)

    print('Linked list after inserting few elements:')
    llist.printList()

    print("\nLinked list After deleting few elements:")
    llist.deleteNode(1)
    llist.deleteNode(4)
    llist.deleteNode(5)
    llist.deleteNode(6)
    
    
    llist.printList()

File: Linked_List/java/linked_list_delete.java
class LinkedList_delete {
    Node head;
  
    class Node {
      int item;
      Node next;
  
      Node(int d) {
        item = d;
        next = null;
      }
    }
  

  
    public void insertAtEnd(int data) {
      Node new_node = new Node(data);
  
      if (head == null) {
        head = new Node(data);
        return;
      }
  
      new_node.next = null;
  
      Node last = head;
      while (last.next != null)
        last = last.next;
  
      last.next = new_node;
      return;
    }
  
    void deleteNode(int position) {
      if (head == null)
        return;
  
      Node node = head;
  
      if (position == 0) {
        head = node.next;
        return;
      }
      for (int i = 0; node != null && i < position - 1; i++)
        node = node.next;
  
      if (node == null || node.next == null)
        return;
  
      Node next = node.next.next;
      node.next = next;
    }
  
    public void printList() {
      Node node = head;
      while (node != null) {
        System.out.print(node.item + " ");
        node = node.next;
      }
    }
  
    public static void main(String[] args) {
      LinkedList_delete llist = new LinkedList_delete();
  
      llist.insertAtEnd(1);
      llist.insertAtEnd(4); 
      llist.insertAtEnd(2);
      llist.insertAtEnd(3);
      llist.insertAtEnd(6);
      llist.insertAtEnd(7);
      llist.insertAtEnd(9);
      llist.insertAtEnd(5);
      llist.insertAtEnd(11);
      llist.insertAtEnd(4);
      System.out.println("Linked list after inserting few elements:");
      llist.printList();
  
      System.out.println("\nLinked list After deleting few elements:");
      llist.deleteNode(1);
      llist.deleteNode(4);
      llist.deleteNode(5);
      llist.deleteNode(6);
      llist.printList();
    }
  }